### DISCUSSION TITLE
Creating gif from two or more than two image

### DISCUSSION THREAD

**Discussion by rajrajsahil** [dlemstra/Magick.NET#1206] [Help] [answered]
```C#
public MagickImageCollection CreateGif(FileStreamResult defectiveImage, FileStreamResult gifSecondImage, int gifTime)
        {
            /*System.IO.MemoryStream gif = new System.IO.MemoryStream();
            GifEncoder encode = new GifEncoder(gif);
            encode.FrameDelay = TimeSpan.FromMilliseconds(gifTime);
            encode.AddFrame(Image.FromStream(defectiveImage.FileStream));
            encode.AddFrame(Image.FromStream(gifSecondImage.FileStream));
            return gif;*/
            defectiveImage.FileStream.Position = 0;
            MagickImage image1= new MagickImage(defectiveImage.FileStream);
            image1.Depth = 1;
            gifSecondImage.FileStream.Position = 0;
            MagickImage image2 = new MagickImage(gifSecondImage.FileStream);
            image2.Depth = 1;

            MagickImageCollection gif = new MagickImageCollection();
            try
            {
                gif.Add(image1);
                gif[0].AnimationDelay = 100;
                gif.Add(image2);
                gif[1].AnimationDelay = 100;
                gif[1].Flip();
                /*gif.Optimize();*/
                return gif;
            }
            catch(Exception ex)
            {
                System.Console.WriteLine("Erro while creatng GIF: {0}", ex);
                throw;
            }
            
        }
```

The gif I get Is just a static image only. According to code the gif should have two image blinking at interval of 1 second

**Comment by dlemstra** [comment]
Can you share the output image and which of the two images are you seeing? And the animation delay that you have now is `100ms` instead of a second.

**Reply by rajrajsahil** [reply]
![4](https://user-images.githubusercontent.com/20652529/176828002-6aec494b-db8b-4d99-9750-1cd4feb8079f.GIF)
This is what I get as GIF. The extension is .GIF but seems to be just a static image. And this is the first image. I tried with 1000ms(1second) also there is no difference

**Comment by rajrajsahil** [comment]
What could be the possible reason for this?

**Reply by dlemstra** [reply]
The GIF that you attached here is a `PNG` file instead?

**Reply by rajrajsahil** [reply]
This file is exactly what I am receiving as the output of CreateGif(..) function above.
Yes as a extension it has .GIF  but it is PNG file. This is what the issue is! I do not see any issue/reason for this in the code.

**Comment by dlemstra** [comment]
How are you calling `gif.Write`? I suspect that you are using a stream without specifying the output format. When you don't specify the format the output format will be the format of the first image.

**Comment by rajrajsahil** [comment]
gif.write() is working fine but I do not have to write it on disk. I have to return it like this - 
            {
                gif.Add(image1);
                gif[0].AnimationDelay = 100;
                gif.Add(image2);
                gif[1].AnimationDelay = 100;
                gif[1].Flip();
                gif.Optimize();
                return gif; // This gif is basically a png only. How to make if GIF
             }

I have other imges/gif and I am putting all in zip file. So I do not have to write on disk
Can you please tell me what need to change in this code for returning Gif

By the way, Thank You so much for you help

**Reply by dlemstra** [reply]
How are you writing/saving the `gif` image?

**Comment by rajrajsahil** [comment]
ZipArchive archive = new ZipArchive(Response.BodyWriter.AsStream(), ZipArchiveMode.Create)                
                       {
                        var entry = archive.CreateEntry(gifname);
                        using (var entryStream = entry.Open())
                        {
                           // This is the one to create gif. This fucntion CreateGif() should return gif which  I am putting into zip. 
                            MagickImageCollection gif= _helper.CreateGif(defectiveImage, gifSecondImage, gifTime); 
                            FileContentResult gifContent = File(gif.ToByteArray(), "image/gif", "mygif.Gif");
                            FileStreamResult gifFile = new FileStreamResult(new MemoryStream(gifContent.FileContents), gifContent.ContentType);
                            gifFile.FileDownloadName = gifContent.FileDownloadName;
                            gifFile.FileStream.Position = 0;
                            await gifFile.FileStream.CopyToAsync(entryStream);
                        }
                    }
Earlier I used different library to create GIF.  This was working fine for both GIF and image.
I shifted to this gif library because I need to run on both linux and windows.

**Comment by dlemstra** [comment, marked as answer]
You should do `gif.ToByteArray(MagickFormat.Gif)` instead.

**Comment by rajrajsahil** [comment]
Thank You very much. It works for me.
